homeassistant/components/opnsense/__init__.py

```diff
@@ -27,6 +33,7 @@
                 vol.Required(CONF_URL): cv.url,
                 vol.Required(CONF_API_KEY): cv.string,
                 vol.Required(CONF_API_SECRET): cv.string,
+                vol.Required(CONF_TIMEOUT, default=20): cv.positive_int,
```

Comment:
For integrations that connect to devices or services, we no longer accept new YAML configuration or changes.

This integration needs to be refactored first to use a configuration flow and config entries first.

More information about this can be found in Architecture Decision Record:

- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

See our developer documentation on how to get started creating a configuration flow for this integration:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

As these changes often involve a bit of work and some significant shift in the current code, we will close this PR for now.

We (and our community!) would really appreciate it if you could start on the refactoring of this integration in a new PR.

Thanks already! :+1:

Reply:
Thanks for the quick review @frenck. This totally makes sense and I am definitely interested in building out a configuration flow for this integration. As I (and it seems like other OPNSense users) are impacted by this timeout I've reopened this PR with just overriding the default timeout and not changing any configuration: https://github.com/home-assistant/core/pull/95314 Please let me know if this is in bad form or not allowed. I just really want my light automations to work again 🤞